lib/core/request.js

```diff
@@ -114,9 +131,16 @@ class Request {
     this.servername = util.getServerName(this.host)
 
     this[kHandler] = handler
+
+    if (channels.create.hasSubscribers) {
+      channels.create.publish({ request: this })
+    }
   }
 
   onBodySent (chunk) {
+    if (channels.bodySent.hasSubscribers) {
+      channels.bodySent.publish({ request: this })
+    }
```

Comment:
@mcollina I don't think this does what you think it does... it get's called for every body chunk?

Reply:
Ah definitely. Have we got a callback for that?

Reply:
Not at the moment. I'm not too happy with the naming here either. But I guess if `chunk === null` that could be end? We need to change that though.